File: Hackerrank/PlusMinus.py
def plus_minus(D, A):
    #print to number of decimals
    #find ratio of pos, neg and 0
    #get sum
    total_len = len(A)
    pos = []
    neg = []
    zero = []
    for x in A:
        if x > 0:
            pos.append(x)
        if x < 0:
            neg.append(x)
        if x == 0:
            zero.append(x)
    posR = len(pos)/total_len
    
    print(format(posR, '.6f'))
    print(format(len(neg)/total_len, '.6f'))
    print(format(len(zero)/total_len, '.6f'))
    
    #print('{0:.%df}'.format(posR)) % (D)
    

D = 6
A = [-4,3,-9,0,4,1]
plus_minus(D, A)

File: Hackerrank/PlusMinus.js
function plus_minus(A){
    let plus = 0
    let minus = 0
    let zero = 0
    for(let i=0; i<A.length; i++){
        if(A[i] < 0) minus += 1
        if(A[i] === 0) zero += 1
        if(A[i] > 0){
            plus += 1
            console.log('in plus', plus)
        } 
    }
    console.log({plus, minus, zero})
    plus = plus/A.length
    minus = minus/A.length
    minus = minus.toFixed(2)
    zero = zero/A.length
    zero = zero.toFixed(2)
    return `plus ${plus}, minus ${minus}, zero ${zero}`
}

A = [-4,3,-9,0,4,1]
console.log(plus_minus(A))